src/app/components/TxHistory/TxHistory.tsx

```diff
@@ -10,8 +11,8 @@ export const TxHistory = ({ history }: { history: TxHistoryEntry[] }) => (
     {history
       .slice()
       .sort((a, b) => b.timestamp - a.timestamp)
-      .map((h, i) => (
-        <React.Fragment key={i}>
+      .map((h) => (
+        <React.Fragment key={generateUUID()}>
```

Comment:
I don't think this will work, since it will re-generate the UUID every time the component is rendered. That may have similar or worse performance issues as compared to using the index.